### ISSUE TITLE
第一次写mvp，按照你那写的但是FrmeMvpHomePresenter.getData()这个报空指针

### ISSUE THREAD

**Issue by Troll4it** [Hankkin/MvpDemo#1]
你好，这是什么问题，求解，谢谢

![1pq0pr 35x nv0vpoh5ae9s](https://user-images.githubusercontent.com/27883219/27813767-b3d6dc18-60aa-11e7-82ae-675201f4afbe.png)

![sv1eu ln92e e n1lbvb h1](https://user-images.githubusercontent.com/27883219/27813667-1360d6f8-60aa-11e7-8831-a35261769ef8.png)

**Comment by Hankkin** [comment]
你好 目测是你的presenter为null 没有初始化

**Comment by Troll4it** [comment]
但是这个初始化不是在这里就初始化了吗？？还需要在哪里初始化
public abstract class MvpActivity<V,P extends BasePresent<V>> extends BaseAcitvity{

    protected P presenter;



    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i("FrameMvpActivity", "onCreate: ");
        presenter = initPresenter();
    }

    @Override
    protected void onResume() {
        super.onResume();
        presenter.attach((V) this);

    }

    @Override
    protected void onDestroy() {
        presenter.detach();
        OkGo.getInstance().cancelTag(this);
        super.onDestroy();
    }

    public abstract P initPresenter();

}

**Comment by Hankkin** [comment]
你的FrameMvpActivity继承了MVPActivity 实现了initPresenter（）抽象方法了，FrameMvpActivity里面初始化你FrameMvpActivity的Presenter啊

**Comment by Troll4it** [comment]
@Override
    public FrmeMvpHomePresenter initPresenter() {
        return new FrmeMvpHomePresenter();
    }这个不是的吗？

**Comment by Hankkin** [comment]
和我的代码对比一下看看哪里不一样，这样我也看不出来

**Comment by Troll4it** [comment]
好的，谢谢

**Comment by Troll4it** [comment]
![image](https://user-images.githubusercontent.com/27883219/27814539-b73484a0-60af-11e7-9ad7-c15493541307.png)
![image](https://user-images.githubusercontent.com/27883219/27814543-bfadff94-60af-11e7-883b-666db310edd2.png)
我只要把你的presenter.getGankData();放到外面也是空指针错误
